statbot/client.py

```diff
@@ -14,283 +14,273 @@
 
 __all__ = [
     'LOG_FULL_MESSAGES',
-    'make_client',
+    'EventIngestionClient',
 ]
 
 LOG_FULL_MESSAGES = False
 
 from .sql import DiscordSqlHandler
 from .util import get_emoji_name, null_logger
 
-def make_client(config, logger=null_logger, sql_logger=null_logger):
-    client = discord.Client()
-    client.ready = False
-    sql = DiscordSqlHandler(config['url'], sql_logger)
+class EventIngestionClient(discord.Client):
 
-    def _accept_message(message):
-        if not client.ready:
-            logger.warn("Can't log message, not ready yet!")
+    def __init__(self, config, logger=null_logger, sql_logger=null_logger):
+        self.config = config
+        self.ready = False
+
+        self.logger = logger
+        self.sql_logger = sql_logger
```

Comment:
You don't need to save `sql_logger`. It's only needed as an argument to `DiscordSqlHandler`